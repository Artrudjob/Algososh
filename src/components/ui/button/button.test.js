import React from "react";
import {create} from "react-test-renderer";
import {render, fireEvent, screen} from "@testing-library/react";
import {Button} from "./button";

describe("TEST BUTTON", () => {
    test("Button with text", () => {
        const componentButton = create(<Button text={"text"} />);
        expect(componentButton.toJSON()).toMatchSnapshot();
    })

    test("Button with not text", () => {
        const componentButton = create(<Button />);
        expect(componentButton.toJSON()).toMatchSnapshot();
    })

    test("Button disabled", () => {
        const componentButton = create(<Button disabled={true} />);
        expect(componentButton.toJSON()).toMatchSnapshot();
    })

    test("Button in loading state", () => {
        const componentButton = create(<Button isLoader={true} />);
        expect(componentButton.toJSON()).toMatchSnapshot();
    })

    test("Calling a callback when the button is clicked.", () => {
        render(<Button />);
        fireEvent.click(screen.getByRole("button"));
    })
})
