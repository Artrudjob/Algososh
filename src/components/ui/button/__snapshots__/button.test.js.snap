// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEST BUTTON Button disabled 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`TEST BUTTON Button in loading state 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`TEST BUTTON Button with not text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`TEST BUTTON Button with text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    text
  </p>
</button>
`;
